src/Microsoft.AspNet.OData.Shared/Formatter/EdmLibHelpers.cs

```diff
@@ -977,14 +990,18 @@ internal static IEdmTypeReference GetExpectedPayloadType(Type type, ODataPath pa
         {
             IEdmTypeReference expectedPayloadType = null;
 
-            if (typeof(IEdmObject).IsAssignableFrom(type))
+            if (typeof(IEdmObject).IsAssignableFrom(type) || typeof(IDeltaSet).IsAssignableFrom(type))
             {
                 // typeless mode. figure out the expected payload type from the OData Path.
                 IEdmType edmType = path.EdmType;
+
                 if (edmType != null)
                 {
                     expectedPayloadType = EdmLibHelpers.ToEdmTypeReference(edmType, isNullable: false);
-                    if (expectedPayloadType.TypeKind() == EdmTypeKind.Collection)
+
+                    //This loop should execute only if its not a type of EdmChangedObjectCollection, In case of EdmChangedObjectCollection,
```

Comment:
```suggestion
                    // This logic should execute only if its not a type of EdmChangedObjectCollection. In case of EdmChangedObjectCollection,
```